Assets/_BForBoss/_Weapons/Scripts/Weapons/WeaponsManager.cs

```diff
@@ -0,0 +1,131 @@
+        [SerializeField]
+        private float _weaponBobFrequency = 10f;
+        [InfoBox("How fast the weapon bob is applied, bigger value is faster")]
+        [SerializeField] 
+        private float _weaponBobSharpness = 10f;
+
+        [Title("Recoil Properties")]
+        [SerializeField] 
+        private float _maxRecoilDistance = 0.5f;
+        [InfoBox("This will affect how fast the recoil moves the weapon, the bigger the value, the fastest")]
+        [SerializeField]
+        private float _recoilSharpness = 50f;
+        [InfoBox("How fast the weapon goes back to it's original position after the recoil is finished")]
+        [SerializeField]
+        private float _recoilRestitutionSharpness = 10f;
+
+        private float _weaponBobFactor = 0;
+        private Vector3 _weaponBobLocalPosition;
+        private Vector3 _accumulatedRecoil;
+        private Vector3 _weaponRecoilLocalPosition;
+
+        public void Initialize(ICharacterMovement characterMovement)
+        {
+            _characterMovement = characterMovement;
+        }
+
+        private void Start()
+        {
+            _equipmentBehaviour.Weapons.ForEach(weapon => weapon.OnFireWeapon += OnWeaponFired);
+        }
+
+        private void OnDestroy()
+        {
+            _equipmentBehaviour.Weapons.ForEach(weapon => weapon.OnFireWeapon -= OnWeaponFired);
+        }
+
+        private void OnWeaponFired(int _)
+        {
+            _accumulatedRecoil += Vector3.back * _equipmentBehaviour.CurrentWeapon.VisualRecoilForce;
+            _accumulatedRecoil = Vector3.ClampMagnitude(_accumulatedRecoil, _maxRecoilDistance);
+        }
+
+        private void OnValidate()
+        {
+            if (_equipmentBehaviour == null)
+            {
+                Debug.LogWarning("Equipment Behaviour missing from WeaponsManager");
+            }
+        }
+
+        private void LateUpdate()
+        {
+            UpdateBobbingWeaponOnMovement();
+            UpdateWeaponRecoil();
+
+            _weaponHolder.transform.localPosition = _weaponBobLocalPosition + _weaponRecoilLocalPosition;
+        }
+
+        private void UpdateBobbingWeaponOnMovement()
+        {
+            var characterVelocity = _characterMovement.CharacterVelocity;
+            var characterMovementFactor = 0f;
+            if (CanBobWeapon())
+            {
+                characterMovementFactor =
+                    Mathf.Clamp01(characterVelocity.magnitude / _characterMovement.CharacterMaxSpeed);
+            }
+            
+            _weaponBobFactor = Mathf.Lerp(_weaponBobFactor, characterMovementFactor, _weaponBobSharpness * Time.deltaTime);
+            
+            var hBobValue = Mathf.Sin(Time.time * _weaponBobFrequency) * _hipFireBobAmount * _weaponBobFactor;
+            /// Trignometric Graph Tranformation: y = A * Sin(b * x - c) + d
+            var vBobValue = (Mathf.Sin(Time.time * _weaponBobFrequency * 2f) * 0.5f + 0.5f) * _weaponBobFactor * _hipFireBobAmount;
```

Comment:
@unoctium1 only part with hardcoded values, since i honestly am not sure how to name them. 
But I'M GUESSING it's for Trig graph transforms

Reply:
Yeah it looks like its just constraining the sine values to be between 0 and 1. I'm fine leaving these ones, it was really just the 0.99 that bothered me